### DISCUSSION TITLE
Using C and Python at the same time

### DISCUSSION THREAD

**Discussion by felipetesc** [electronstudio/raylib-python-cffi#96] [Ideas]
Is it possible to use Python with C in a way that, for example, we can create some parts of the game, or the application, using c for better performance, and in some other parts, which aren't so 'heavy',with Python. How could I achieve that, that's, can we do that ? For instance, I'm thinking about using Python to draw the UI stuff, and for the rendering process of 3D objects to be programmed(rendered) using C code. 
What changes would be needed to this project to connect your library to C source code? Thanks !

**Comment by electronstudio** [comment]
What you describe is how all large game engines are written. The engine is written in C(++) and then a scripting language is embedded and used to write the game logic. Usually it's C#, Lua, Gdscript because they are better suited but the Python interpreter can be embedded in a C program. 

The other way of doing it (that isn't commonly done) is to write the main program in Python and then have it load C modules. Lookup Python ctypes, CFFI, etc. 

If you just want to use Python with better performance then look up Nuitka, MypyC and PyPy.

**Reply by felipetesc** [reply]
Thanks for your reply.  The reason for this question/discussion is based on the information presented at : https://electronstudio.github.io/raylib-python-cffi/README.html#how-to-use. There, it is provided a comparison between C and Pyhton. The Bunnymark with CFFI 3.7 Python is 80% 'slower' (not sure if slower is the correct word, or even to say 80% slower) than its C counterpart.

That said, I think this binding to raylib to be the best one I found so far (used Rust, Haxe, Odin,...). 

PS: Thanks for this project
